src/couch/src/couch_db.erl

```diff
@@ -742,18 +742,16 @@ is_authorized(#user_ctx{name = UserName, roles = UserRoles}, Security) ->
         false -> check_security(names, UserName, Names)
     end.
 
-check_security(roles, [], _) ->
-    false;
-check_security(roles, UserRoles, Roles) ->
+check_security(roles, [_ | _] = UserRoles, [_ | _] = Roles) ->
     UserRolesSet = ordsets:from_list(UserRoles),
     RolesSet = ordsets:from_list(Roles),
     not ordsets:is_disjoint(UserRolesSet, RolesSet);
-check_security(names, _, []) ->
-    false;
-check_security(names, null, _) ->
+check_security(roles, _, _) ->
     false;
-check_security(names, UserName, Names) ->
-    lists:member(UserName, Names).
+check_security(names, UserName, [_ | _] = Names) ->
+    lists:member(UserName, Names);
+check_security(names, _, _) ->
```

Comment:
One behavior difference it seems is previously `check_security(names, null, [<<"foo">>, null])` would have returned `false` but with the new code it would return `true`. However, I don't know how realistic is to have `null` in the list of `Names`, perhaps it's even impossible but since it's the authorization code we should triple check that. (cc:  @janl / @rnewson what do you think?)

Reply:
The discussion, which way to go starts [here](https://couchdb.slack.com/archives/C01TBE2J197/p1655505440919899). So we can also react in `jwt_authentication_handler/1` in combination with the open and ready for review PR #4041